### PR TITLE
SLING-7194 AdapterManager sorts AdapterFactory implementations lowest ranking first

### PR DESCRIPTION
**Pull request by stefanseifert**

### CODE REVIEW

File: src/main/java/org/apache/sling/adapter/internal/AdapterFactoryDescriptorMap.java [hunk 1]
```diff
@@ -41,4 +42,16 @@
 
     private static final long serialVersionUID = 2L;
 
+    public AdapterFactoryDescriptorMap() {
+        super(new Comparator<ServiceReference<AdapterFactory>>() {
```
Comment (kwin) [inline src/main/java/org/apache/sling/adapter/internal/AdapterFactoryDescriptorMap.java]: Can you just use https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html#reversed--?

Comment (stefanseifert) [reply src/main/java/org/apache/sling/adapter/internal/AdapterFactoryDescriptorMap.java]: this seems to be java8 only?

Comment (kwin) [reply src/main/java/org/apache/sling/adapter/internal/AdapterFactoryDescriptorMap.java]: Sorry, you are right. I guess in this module we still need to support older Java versions.

**Comment by stefanseifert** [comment]
this PR is obsolete - as discussed in SLING-7194 we do not change the existing behavior but document it better

**Pull request closed without merging**